code/edge/src/main/java/com/adobe/marketing/mobile/EdgeHitProcessor.java

```diff
@@ -302,13 +304,22 @@ private Map<String, String> getRequestHeaders() {
 		}
 
 		// get latest Assurance shared state
-		final Map<String, Object> assuranceSharedState = sharedStateCallback.getSharedState(
+		SharedStateResult assuranceStateResult = sharedStateCallback.getSharedState(
 			EdgeConstants.SharedState.ASSURANCE,
 			null
 		);
-		final String assuranceIntegrationId = EventUtils.getAssuranceIntegrationId(assuranceSharedState);
 
-		if (!Utils.isNullOrEmpty(assuranceIntegrationId)) {
+		if (assuranceStateResult == null || assuranceStateResult.getStatus() != SharedStateStatus.SET) {
+			return requestHeaders;
+		}
+
+		final String assuranceIntegrationId = DataReader.optString(
+			assuranceStateResult.getValue(),
+			EdgeConstants.SharedState.Assurance.INTEGRATION_ID,
+			null
+		);
+
+		if (!StringUtils.isNullOrEmpty(assuranceIntegrationId)) {
```

Comment:
I've only updated EdgeHitProcessor to fix uses of getSharedState and to use the new DataReader in this `getRequestHeader` function. The reset of this class will be updated in another PR.